tests/test_blackoil_amg.cpp

```diff
@@ -289,30 +293,18 @@ void runBlackoilAmgLaplace()
     setBoundary(x, b, N, comm.indexSet());
 
     Operator fop(mat, comm);
-    typedef Dune::Amg::CoarsenCriterion<Dune::Amg::SymmetricCriterion<BCRSMat,Dune::Amg::FirstDiagonal> >
-        Criterion;
-    typedef Dune::SeqSSOR<BCRSMat,Vector,Vector> Smoother;
-    //typedef Dune::SeqJac<BCRSMat,Vector,Vector> Smoother;
-    //typedef Dune::SeqILU0<BCRSMat,Vector,Vector> Smoother;
-    //typedef Dune::SeqILUn<BCRSMat,Vector,Vector> Smoother;
-    typedef Dune::BlockPreconditioner<Vector,Vector,Communication,Smoother> ParSmoother;
-    typedef typename Dune::Amg::SmootherTraits<ParSmoother>::Arguments SmootherArgs;
-
     Dune::OverlappingSchwarzScalarProduct<Vector,Communication> sp(comm);
-
     Dune::InverseOperatorResult r;
-    SmootherArgs smootherArgs;
-    Criterion criterion;
-
-    smootherArgs.iterations = 1;
-    Opm::CPRParameter param;
-
-    Opm::BlackoilAmg<Operator,ParSmoother,Criterion,Communication,0,0> amg(param,
-                                                                         {},
-                                                                         fop, criterion,
-                                                                         smootherArgs,
-                                                                         comm);
-    Dune::CGSolver<Vector> amgCG(fop, sp, amg, 10e-8, 300, (ccomm.rank()==0) ? 2 : 0);
+
+    boost::property_tree::ptree prm;
+    prm.put("type", "amg");
+    std::function<Vector()> weights = [&mat]() {
+        return Opm::Amg::getQuasiImpesWeights<BCRSMat, Vector>(mat, 0, false);
+    };
+    auto amg = Opm::PreconditionerFactory<Operator, Communication>::create(fop, prm, weights, comm);
```

Comment:
This looks a bit strange and is hard to understand. At first I thought we would actually set up the weights, but the function is of course never called. Why not make that explicit?
```
auto dummy_weights = [](){return Vector();}
auto amg = Opm::PreconditionerFactory<Operator, Communication>::create(fop, prm, dummy_weights, comm);
```

Reply:
Don't bother. I did not see that this is just in a test.